Use temp allocator for kernel registry

JakeStevens opened the pull request:
Summary: When indexing to the registry to get the op, memory is allocated from the method allocator to instantiate some TensorMeta and included objects. This memory is only used for that purpose and is not needed for the entire lifetime of the Method. Thus, we can instead use temp allocator which can later be reset and free up memory as needed.

Differential Revision: D79285675




cc @larryliu0820 @JacobSzwejbka @lucylq

larryliu0820 reviewed (approved):
Thank you for doing this.

JakeStevens committed:
Use temp allocator for kernel registry (#13012)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/13012

When indexing to the registry to get the op, memory is allocated from the method allocator to instantiate some TensorMeta and included objects. This memory is only used for that purpose and is not needed for the entire lifetime of the Method. Thus, we can instead use temp allocator which can later be reset and free up memory as needed.

Reviewed By: larryliu0820

Differential Revision: D79285675

JakeStevens commented:
Failing tests pre-existing. moshi from undefined libtorchaudio symbol, openvino from backend import. Landing.
